### ISSUE TITLE
Unable to start server: a height hint greater than 0 must be provided

### ISSUE THREAD

**Issue by cyclopes2011** [lightningnetwork/lnd#3539]
### Background

Attempting to restore LND from wallet seed and channel.backup file.

### Your environment

* lnd version 0.7.1-beta commit=v0.7.1-beta-352-9b1ecbd3fa3fd93a6d95a5f71921c73777340760
* 16.04.1-Ubuntu
* Bitcoin Core Daemon version v0.16.3.0-g49e34e288 (installed on dedicated server)

### Steps to reproduce

Performed the steps to restore LND from wallet seed and channel.backup file via `lncli create` command.

### Expected behaviour

Existing wallet restored via `lncli create` and funds returned to wallet after channel restore and closeall per the SCB recovery process.

### Actual behaviour

LND fails to start due to the following error:

`2019-09-24 03:55:09.062 [ERR] LTND: Unable to start server: a height hint greater than 0 must be provided`

Complete startup log attached below:
[20190924 1026 steropes.log](https://github.com/lightningnetwork/lnd/files/3647692/20190924.1026.steropes.log)

**Comment by cyclopes2011** [comment]
Similar to #3444

**Comment by wpaulino** [comment]
The following patch should allow you to start your node again:

```diff --git a/chanrestore.go b/chanrestore.go
index f771f91b..34f74f69 100644
--- a/chanrestore.go
+++ b/chanrestore.go
@@ -138,7 +138,8 @@ func (c *chanDBRestorer) RestoreChansFromSingles(backups ...chanbackup.Single) e
 	for _, restoredChannel := range channelShells {
 		err := c.chainArb.WatchNewChannel(restoredChannel.Chan)
 		if err != nil {
-			return err
+			continue
+			//return err
 		}
 	}
 
```

Closing the issue as a duplicate of #3444.

**Comment by cyclopes2011** [comment]
Here is what I did but the error remains the same.

Sorry, I am not a developer so I assume the @@ is a location for the code function and not part of the edit. I have edited the "chanrestore.go" file per your instruction and have included the comments prior to the section to be edited for clarity:

        // Finally, we'll need to inform the chain arbitrator of these new
        // channels so we'll properly watch for their ultimate closure on chain
        // and sweep them via the DLP.
        for _, restoredChannel := range channelShells {
                err := c.chainArb.WatchNewChannel(restoredChannel.Chan)
                if err != nil {
                        continue
                        // return err
                }
        }

Thank you for your response and assistance. :)

**Comment by wpaulino** [comment]
Did you reinstall lnd with the new changes?

**Comment by cyclopes2011** [comment]
I simply edited the chanrestore.go file and restarted LND. Do I need to do a clean install of LND, the SCB restore, and then start LND?

**Comment by wpaulino** [comment]
You'll need to run `make install` before restarting.

**Comment by cyclopes2011** [comment]
Same error after the make install and restart.

**Comment by wpaulino** [comment]
Are you using the lnd binary found in `$GOPATH/bin/`?

**Comment by cyclopes2011** [comment]
I assume so since I have it in my path. I don't have any other LND installed.

**Comment by wpaulino** [comment]
Could you try this as well?

```diff --git a/contractcourt/chain_arbitrator.go b/contractcourt/chain_arbitrator.go
index b3cd497d..b1e934b9 100644
--- a/contractcourt/chain_arbitrator.go
+++ b/contractcourt/chain_arbitrator.go
@@ -503,7 +503,7 @@ func (c *ChainArbitrator) Start() error {
 	for err := range watcherErrs {
 		if err != nil {
 			c.Stop()
-			return err
+			//return err
 		}
 	}
 
```

**Comment by cyclopes2011** [comment]
(Sorry for the delay.)

Yes! LND is running after that last change. I am going to monitor and let it do its thing.

Thank you for the assistance! :)

**Comment by cyclopes2011** [comment]
LND has been running for a while now. I have manually closed the channels with `lncli closeallchannels -force` but have five channels remaining. Those five produce an error and appear to be stuck and cannot be closed. Three of the five channels have a recurring error in the log:

`2019-09-24 21:58:52.799 [ERR] NANN: Unable to retrieve chan status for Channel(b0f7a019b4d0d4dd48e7071d10c9086a4ce675fa0970a43dc6263e1384202c7c:0): edge not found
2019-09-24 21:58:52.799 [ERR] NANN: Unable to retrieve chan status for Channel(3ebfb564ccd84c643f9b2cc6cf03837d6037f3d87314b623aafb962c5ff2d9a3:0): edge not found
2019-09-24 21:58:52.799 [ERR] NANN: Unable to retrieve chan status for Channel(bbc2e6bf0f1dfd166238df8050ba9cd1c3bf98e267de9db624b38ae2b08f11f4:0): edge not found`

Not sure about the other two channels. There is not an error being logged that I see for these two so I'm looking into more detail on the issue.

I searched the LND issues to find something similar and don't see anything directly applicable. Let me know if you have any advice on how to resolve these channels. Thank you!

I can open a new issue if needed and provide more details in the morning.
